src/System.Text.Json/src/JsonReflectionEmitMaterializer.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Reflection;
+using System.Reflection.Emit;
+
+namespace System.Text.Json.Serialization
+{
+    internal class JsonReflectionEmitMaterializer : JsonMemberBasedClassMaterializer
+    {
+        public override JsonClassInfo.ConstructorDelegate CreateConstructor(Type type)
+        {
+            ConstructorInfo realMethod = type.GetConstructor(BindingFlags.Public | BindingFlags.NonPublic| BindingFlags.Instance, binder: null, Type.EmptyTypes, modifiers: null);
+            if (realMethod == null)
+                return null;
+
+            var dynamicMethod = new DynamicMethod(
+                realMethod.Name,
+                type,
+                Type.EmptyTypes,
+                typeof(JsonReflectionEmitMaterializer).Module,
+                skipVisibility: true);
+
+            if (dynamicMethod != null)
+            {
+                ILGenerator generator = dynamicMethod?.GetILGenerator();
```

Comment:
The line above already checked that dynamicMethod isn't null.  The "?." can just be ".".